src/fs-cache.js

```diff
@@ -125,35 +125,36 @@ module.exports = function(params, callback) {
   const identifier = params.identifier;
   let directory;
 
-  if (typeof params.directory === "string") {
-    directory = params.directory;
-  } else {
-    directory = findCacheDir({ name: "babel-loader" }) || os.tmpdir();
+  try {
+    if (typeof params.directory === "string") {
+      directory = params.directory;
+      mkdirp.sync(directory);
+    } else {
+      directory = findCacheDir({ name: "babel-loader", create: true });
```

Comment:
Only if it can find a parent directory with `package.json`: https://github.com/avajs/find-cache-dir/blob/7df8ba743347945d1f6d51a281fe6776d656b104/index.js#L17-L19

If `pkgDir.sync(dir)` returns `null`, `findCacheDir` will also return `null` regardless of the `create` option.

+1 for adding a test case.